### PR TITLE
Cover some notification commands by Unit Tests

### PR DESCRIPTION
**Pull request by LevchenkoS**
Has been covered:
- `OnTBTClientStateNotification`;
- `OnTouchEventNotification`;
- `OnVehicleDataNotification`;
- `OnWayPointChangeNotification`.

Related to: [APPLINK-25925](https://adc.luxoft.com/jira/browse/APPLINK-25925)

@AByzhynar, @okozlovlux, @Kozoriz, @LuxoftAKutsan, @vlantonov, @wolfylambova22, @VVeremjova , @pvvasilev, please review.

### CODE REVIEW

File: src/components/application_manager/test/commands/mobile/on_tbt_client_state_notification_test.cc [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  const bool kIsProtocolVersionCorrect =
+      (*arg)[am::strings::params][am::strings::protocol_version].asInt() ==
+      am::commands::CommandImpl::protocol_version_;
+
+  const bool kIsNotificationCorrect =
+      (*arg)[am::strings::params][am::strings::message_type].asInt() ==
+      am::MessageType::kNotification;
+
+  const bool kIsConnectionKeyCorrect =
+      (*arg)[am::strings::params][am::strings::connection_key] == kAppId;
```
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/on_tbt_client_state_notification_test.cc]: @LevchenkoS why do you use **asInt()** in previous assigments and in this don't?

File: src/components/application_manager/test/commands/mobile/on_touch_event_notification_test.cc [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdint.h>
+#include <string>
```
Comment (okozlovlux) [inline src/components/application_manager/test/commands/mobile/on_touch_event_notification_test.cc]: @LevchenkoS where do you use string in this test?

File: src/components/application_manager/test/commands/mobile/on_tbt_client_state_notification_test.cc [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+  return kIsMobileProtocolTypeCorrect && kIsProtocolVersionCorrect &&
+         kIsNotificationCorrect && kIsConnectionKeyCorrect;
+}
+
+TEST_F(OnTBTClientStateNotificationTest, Run_SUCCESS) {
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/on_tbt_client_state_notification_test.cc]: @LevchenkoS Think about adding more information into test case name. I mean incoming parameters.

**Comment by Kozoriz** [comment]
Reviewed

**Comment by vlantonov** [comment]
@LevchenkoS Reviewed

**Comment by pvvasilev** [comment]
Reviewed.

File: src/components/application_manager/test/commands/mobile/on_tbt_client_state_notification_test.cc (hunk 3 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/on_tbt_client_state_notification_test.cc]: @AByzhynar, fixed in 61ac019.

File: src/components/application_manager/test/commands/mobile/on_tbt_client_state_notification_test.cc (hunk 1 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/on_tbt_client_state_notification_test.cc]: @okozlovlux, fixed in 61ac019.

File: src/components/application_manager/test/commands/mobile/on_touch_event_notification_test.cc (hunk 2 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/on_touch_event_notification_test.cc]: @okozlovlux, fixed in 61ac019.

**Comment by okozlovlux** [comment]
@LevchenkoS , reviewed

**Comment by VVeremjova** [comment]
Reviewed

File: src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+  const bool kIsNotificationCorrect =
+      (*arg)[am::strings::params][am::strings::message_type].asInt() ==
+      am::MessageType::kNotification;
+
+  const bool kIsConnectionKeyCorrect =
+      (*arg)[am::strings::params][am::strings::connection_key].asUInt() ==
+      kAppId;
+
+  const bool kIsMsgParamsCorrect =
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc]: @LevchenkoS  Are `kAreMsgParamsCorrect`

File: src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      (*arg)[am::strings::msg_params][key].asInt() == value;
+
+  return kIsMobileProtocolTypeCorrect && kIsProtocolVersionCorrect &&
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc]: @LevchenkoS Try to use helpers::Compare

**Comment by anosach-luxoft** [comment]
@LevchenkoS please stop using `kCamelCase` template for local variables naming. We use common naming convention for const and non-const local variables so please use it. If you with @AByzhynar have thoughts that we need to change approach please raise this question in proper way.

File: src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc (hunk 5 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc]: @AByzhynar, fixed in 1397fcf.

File: src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc (hunk 4 above)
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/on_vehicle_data_notification_test.cc]: @AByzhynar, fixed in 1397fcf.

**Comment by AByzhynar** [comment]
Reviewed

**Pull request merged**